libblkid/src/superblocks/luks.c

```diff
@@ -34,6 +34,8 @@
 #define LUKS_MAGIC	"LUKS\xba\xbe"
 #define LUKS_MAGIC_2	"SKUL\xba\xbe"
 
+#define LUKS2_HW_OPAL_SUBSYSTEM	"HW-OPAL"
```

Comment:
This must be a C string, you must chcek also trailing \0. (With memcmp just add it here.)

Reply:
This acts as a C-string. `printf("%u\n", (unsigned) sizeof(LUKS2_HW_OPAL_SUBSYSTEM));` outputs `8`, which is correct.